### ISSUE TITLE
Add support for `:doc:` role

### ISSUE THREAD

**Issue by humitos** [readthedocs/sphinx-hoverxref#18]
I think it's not hard to implement showing tooltips on `:doc:`. Although, this will retrieve the whole page/document and may be "too much" content to be displayed inside the tooltip.

Once the full doc is retrieved, we could cut its content after the first title breaking (`h1` or `h2`, for example).

**Comment by astrojuanlu** [comment]
+1! I was wondering if this was a misconfiguration on my side, glad that you're tracking it.

**Comment by astrojuanlu** [comment]
About "too much content", I think sphinx-hoverxref handles this nicely already. See for instance the `API` tooltip on the poliastro index: https://docs.poliastro.space/en/stable/

![Screenshot 2021-11-25 at 09-13-01 poliastro - Astrodynamics in Python — poliastro 0 15 2 documentation](https://user-images.githubusercontent.com/316517/143404116-3c830825-6460-4cd8-8bc9-132a7e663504.png)

It's a `{ref}` for a whole document, so I could be using `{doc}` perfectly instead.

Maybe a first iteration loading the whole page and then let people complain? :)

**Comment by humitos** [comment]
> About "too much content", I think sphinx-hoverxref handles this nicely already. See for instance the `API` tooltip on the poliastro index: [docs.poliastro.space/en/stable](https://docs.poliastro.space/en/stable/)

This is correct. I think when I opened this issue the modal (which is a completely different js library) was not implemented yet.

Also, we can now make explicit on what roles we want tooltips. So, users can disable on `:doc:` if they don't want to use a different style (modal), for example.

> Once the full doc is retrieved, we could cut its content after the first title breaking (`h1` or `h2`, for example).

I tried to do this at the backend level and I failed --but I was doing a lot of things all together at that time (https://github.com/readthedocs/readthedocs.org/pull/8319#issuecomment-875709891). Maybe we can give it a try again.

**Comment by ericholscher** [comment]
@humitos Is this hard to do? I keep wishing we had it, and I'd love to see a simple implementation if it's not a huge pain.

**Comment by humitos** [comment]
@ericholscher I don't think it should be hard to do the simple version: "Return all the page".

I'm adding this issue to the roadmap in Q1, so we prioritize it soon.

**Comment by ericholscher** [comment]
The new implementation will likely change this design, so holding off on this 👍
